packages/caliper-fabric/test/connector-versions/peer-gateway/PeerGateway.js

```diff
@@ -0,0 +1,440 @@
+            });
+
+            it('when no contractId provided', async () => {
+                const request = {
+                    channel: 'mychannel',
+                    contractId: '',
+                    contractFunction: '',
+                    invokerIdentity: 'admin'
+                };
+                await fabricGateway._sendSingleRequest(request).should.be.rejectedWith(/No contractId provided/);
+                request.contractId = null;
+                await fabricGateway._sendSingleRequest(request).should.be.rejectedWith(/No contractId provided/);
+                delete request.contractId;
+                await fabricGateway._sendSingleRequest(request).should.be.rejectedWith(/No contractId provided/);
+            });
+
+            it('when channel provided but contractId is not a valid chaincode id', async () => {
+                const request = {
+                    channel: 'yourchannel',
+                    contractId: 'findingMyMarbles',
+                    contractFunction: 'myFunction',
+                    invokerIdentity: 'admin',
+                };
+                await fabricGateway._sendSingleRequest(request).should.be.rejectedWith(/Unable to find specified contract findingMyMarbles on channel yourchannel/);
+            });
+
+            it('when no channel provided and contract id is not a valid contract id', async () => {
+                const request = {
+                    channel: '',
+                    contractId: 'not-a-valid-contract-id',
+                    contractFunction: '',
+                    invokerIdentity: 'admin',
+                };
+                await fabricGateway._sendSingleRequest(request).should.be.rejectedWith(/Could not find details for contract ID not-a-valid-contract-id/);
+            });
+        });
+
+        describe('when making a submit request', () => {
+            it('should succeed if status is successful and return with an appropriate TxStatus', async () => {
+                const args = ['arg1', 'arg2'];
+                const request = {
+                    channel: 'mychannel',
+                    contractId: 'marbles',
+                    contractFunction: 'myFunction',
+                    contractArguments: args,
+                    invokerIdentity: 'admin',
+                };
+                const txStatus = await fabricGateway._sendSingleRequest(request);
+                txStatus.should.be.instanceOf(TxStatus);
+                txStatus.GetID().should.equal('1');
+                txStatus.GetStatus().should.equal('success');
+                txStatus.GetResult().should.equal('submitResponse');
+                txStatus.IsVerified().should.be.true;
+
+                Transaction.submit.should.be.true;
+                Transaction.submitArgs.should.deep.equal(args);
+                Transaction.constructorArgs.should.equal('myFunction');
```

Comment:
In the other tests there is this block

```
                Transaction.reset();

                request.readOnly = false;
                await fabricGateway._sendSingleRequest(request);
                Transaction.submit.should.be.true;
                Transaction.submitArgs.should.deep.equal(args);
                Transaction.constructorArgs.should.equal('myFunction');
```
this is testing when readOnly is explictly set to false which should be included